numpy/_core/include/numpy/dtype_api.h

```diff
@@ -48,6 +50,12 @@ struct PyArrayMethodObject_tag;
         void *reserved[3];
     } PyArray_DTypeMeta;
 
+#else
+
+typedef PyTypeObject PyArray_DTypeMeta;
+
+#endif /* Py_LIMITED_API */
+
```

Comment:
@seberg there are some comments below about "expected changes" for the ArrayMethod API. Do you still want to eventually do those changes? Now is probably the time...